Pxc 4607

Tusamarco opened the pull request:
Fix for  PXC-4607 (synchronize clone plugin persistency)

Tusamarco committed:
removing new line

Tusamarco committed:
Fixing mtr 1

Tusamarco committed:
Fixing mtr 2
Align privileges assignment as for mysql_user

Tusamarco committed:
Synchronise donor - joiner clone plugin persistency

Tusamarco commented:
@ParveezBaig @kamil-holubicki can you check it please ? this should fix the 4067

kamil-holubicki commented on scripts/wsrep_sst_clone.sh:
```diff
@@ -358,12 +358,14 @@ setup_clone_plugin()
         fi
         $MYSQL_ACLIENT -e "${WSREP_OFF}INSTALL PLUGIN CLONE SONAME 'mysql_clone.so';"
-        CLEANUP_CLONE_PLUGIN="yes"
+        if [ "$ROLE" == "donor" ]; then
```
It is not consistent across the script how we do the comparison. Sometimes it is `==` and sometimes it it `=` (just 4 lines above). 
`=` is posix compatible and should be used with single brackets `[...]`
`==` is bash only compatible and should be used with double brackets `[[...]]`

Would it be possible to have it consistent?

Tusamarco replied:
will review all in a different branch/pr

kamil-holubicki commented on scripts/wsrep_sst_clone.sh:
```diff
@@ -392,7 +394,7 @@ setup_clone_plugin()
     [ "$SERVER_SSL_CA" = "NULL" ] && SERVER_SSL_CA=
 
-    if [ "$CLONE_SSL_CERT" = "NULL" -o "$CLONE_SSL_KEY" = "NULL" ]
+    if [ "$CLONE_SSL_CERT" == "NULL" -o "$CLONE_SSL_CERT" = "" ]
```
Mixture of single `=` and double `==` in one `if`

Tusamarco committed:
We have an issue when not using jemalloc in correctly reading this section of the code:
DROP USER IF EXISTS 'mysql.pxc.sst.user'@'localhost'; $CLEANUP_CLONE_PLUGIN_SQL

Just moving the variable in different line fix the issue. this is quick and dirty fix but it works.

Tusamarco committed:
resolving conflicts with PXC-4469
